### PR TITLE
✏️ Studio Event creation

### PR DESCRIPTION
**Pull request by JUtupe**

### CODE REVIEW

File: src/components/ui/PrimaryButton/PrimaryButton.tsx [hunk 1]
```diff
@@ -3,12 +3,14 @@ import React from "react";
 import styles from "./PrimaryButton.module.css";
 
 interface PrimaryButtonProps {
+    type?: 'submit' | 'reset' | 'button';
     onClick?: () => void;
 }
 
-export const PrimaryButton: React.FC<React.PropsWithChildren<PrimaryButtonProps>> = ({ children, onClick }) => {
+export const PrimaryButton: React.FC<React.PropsWithChildren<PrimaryButtonProps>> = ({ children, type = undefined, onClick }) => {
```
Comment (DiD3n) [inline src/components/ui/PrimaryButton/PrimaryButton.tsx]: undefined is default

File: src/components/ui/StepView/StepView.module.scss [hunk 2]
```diff
@@ -0,0 +1,85 @@
+.steps {
+  display: flex;
+
+  flex-direction: row;
+
+  align-items: center;
```
Comment (DiD3n) [inline src/components/ui/StepView/StepView.module.scss]: no need to separate these, it's flex and its properties

File: src/components/ui/StepView/StepView.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    return (
+        <div className={styles.clickableSteps}>
+            {steps.map((step, i) => {
+                return <div key={i.toString()} onClick={() => onStepClicked(i)} className={styles.clickableStepContainer}>
+                    {singleStep(i + 1, step, i === activeStepIndex)}
+                </div>
+            })}
+        </div>
+    )
+}
```
Comment (DiD3n) [inline src/components/ui/StepView/StepView.tsx]: it doesn't seems to be used anywhere

File: src/components/ui/StepView/StepView.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            <span className={cs(styles.index, {[styles.indexActive]: isSelected})}>
+                {index}
+            </span>
+
+            <span className={cs(styles.text, {[styles.textActive]: isSelected})}>
+                {title}
+            </span>
+        </div>
+    )
+}
```
Comment (DiD3n) [inline src/components/ui/StepView/StepView.tsx]: Why not make it proper component?

File: src/components/ui/StepView/StepView.tsx (hunk 3 above)
Comment (DiD3n) [reply src/components/ui/StepView/StepView.tsx]: i think we can avoid making clickable copy-paste of original by moving creating steps to view itself
ex.
```jsx
<StepsList>
   <Step label={firstStep} active={step == 0}>
   <Step label={secoundStep} active={step == 1}>
   <Step label={otherStep} active={step == 2}>
<StepsList/>
```
then making onClick prop or wapper would be easy as hell

File: src/components/ui/StepView/StepView.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+export const StepView: React.FC<StepViewProps> = ({steps, activeStepIndex}) => {
+    return (
+        <div className={styles.steps}>
+            {steps.map((step, i) => {
+                const shouldPlaceArrow = i < steps.length - 1
+
+                return <div key={i.toString()} className={styles.stepContainer}>
+                    {singleStep(i + 1, step, i === activeStepIndex)}
+                    {shouldPlaceArrow && <NextArrow width={28} height={28}/>}
```
Comment (DiD3n) [inline src/components/ui/StepView/StepView.tsx]: u can place arrow in `::before` in css
then removing the last or first arrow would be easy as `:last-child::before { display: none }`

File: src/views/studio/StudioEventCreate/BasicInfo/StudioEventBasicInfo.module.scss [hunk 6]
```diff
@@ -0,0 +1,15 @@
+.row {
+  display: flex;
+
+  justify-content: space-between;
+
+  width: 100%;
+
+  flex-direction: row;
```
Comment (DiD3n) [inline src/views/studio/StudioEventCreate/BasicInfo/StudioEventBasicInfo.module.scss]: to much enters

File: src/views/studio/StudioEventCreate/Lectures/EventLecturesForm.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import PrimaryButton from "../../../../components/ui/PrimaryButton/PrimaryButton";
+
+interface EventLecturesFormProps {
+    eventId: string,
+    onSubmitted: () => void
+}
+
+export const EventLecturesForm: React.FC<EventLecturesFormProps> = ({eventId, onSubmitted}) => {
+    return (
+        <div>
```
Comment (DiD3n) [inline src/views/studio/StudioEventCreate/Lectures/EventLecturesForm.tsx]: is this div necessary?

File: src/views/studio/StudioEventCreate/Theme/EventThemeForm.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import PrimaryButton from "../../../../components/ui/PrimaryButton/PrimaryButton";
+
+interface EventThemeFormProps {
+    eventId: string,
+    onSubmitted: () => void
+}
+
+export const EventThemeForm: React.FC<EventThemeFormProps> = ({eventId, onSubmitted}) => {
+    return (
+        <div>
```
Comment (DiD3n) [inline src/views/studio/StudioEventCreate/Theme/EventThemeForm.tsx]: this too

**Review by DiD3n — CHANGES_REQUESTED** [review]

File: src/components/ui/StepView/StepView.tsx (hunk 3 above)
Comment (JUtupe) [reply src/components/ui/StepView/StepView.tsx]: I'll give it a try

File: src/components/ui/PrimaryButton/PrimaryButton.tsx (hunk 1 above)
Comment (JUtupe) [reply src/components/ui/PrimaryButton/PrimaryButton.tsx]: don

**Review by DiD3n — APPROVED** [review]
legit enough

**Pull request merged**